app/javascript/packs/order_screen.js

```diff
@@ -200,9 +202,8 @@ document.addEventListener('turbolinks:load', () => {
             }
           }).then((response) => {
             this.$set(this.users, this.users.indexOf(this.selectedUser), response.body.user);
-            this.$emit('updateusers');
 
-            if(!this.keepUserSelected) {
+            if(!this.keepUserSelected && this.orderRows.length === 0){
               this.setUser(null);
```

Comment:
I'm a bit weirded out by the way `setUser` is being used. Its responsibility slightly extends beyond what its name suggests

Reply:
True, might be due for a refactor